### ISSUE TITLE
(🐞) Sus impostor `tuple` in dict literal error message

### ISSUE THREAD

**Issue by KotlinIsland** [python/mypy#15973]
```py
from typing import Any

any: Any
{"a": any}  # error: Expression type contains "Any" (has type "tuple[str, Any]")  [misc]
```

`tuple[str, Any]`

**Comment by sobolevn** [comment]
I cannot reproduce this error with default settings, but I think that you mean `--disallow-any-expr`: 

```python
» mypy ex.py --disallow-any-expr
ex.py:4: error: Expression type contains "Any" (has type "tuple[str, Any]")  [misc]
ex.py:4: error: Expression has type "Any" 
```

So, the problem is that the error message is shown twice, isn't it?

**Comment by KotlinIsland** [comment]
> So, the problem is that the error message is shown twice, isn't it?

Well, I can see two expressions that contain any, `any` and `{"a": any}`, but neither of these expressions are a `tuple`.

**Comment by sobolevn** [comment]
When checking `{"a": any}` mypy represents it as a `tuple` of key and value, which is kinda what it is.

**Comment by KotlinIsland** [comment]
> When checking `{"a": any}` mypy represents it as a `tuple` of key and value, which is kinda what it is.

Yeah, that does make sense that mypy would, as an implementation detail, model this as a `tuple`. But in reality it's not, so the error message here doesn't make `Any` sense.

**Comment by popsiclelmlm** [comment]
I checked this against current `master` (`e0c375a97`) and it looks fixed now.

Reproduction I ran:

```bash
python -m mypy --disallow-any-expr repro.py
```

```python
from typing import Any

any: Any
{"a": any}
```

Current output:

```text
repro.py:4: error: Expression has type "Any"  [misc]
Found 1 error in 1 file (checked 1 source file)
```

The internal `tuple[str, Any]` wording from the original report no longer appears. This issue may be ready to close unless there is another remaining reproducer.
